src/libraries/System.Collections.Immutable/src/System/Collections/Frozen/String/KeyAnalyzer.cs

```diff
@@ -129,7 +109,7 @@ private static bool TryUseSubstring(ReadOnlySpan<string> uniqueStrings, bool ign
                 foreach (string s in uniqueStrings)
                 {
                     // Get the span for the substring.
-                    ReadOnlySpan<char> substring = getSubstringSpan(s, index, count);
+                    ReadOnlySpan<char> substring = count == 0 ? s.AsSpan() : Slicer(s, index, count);
```

Comment:
sadly, since we're storing the ENTIRE string in the hashset, we actually need to reslice the string again here... one further tweak would be to make the set we're adding-to request the substrings being inserted instead, but that brings back all that allocation nastiness for the substring-strings which the original PRs were making huge gains by eliminating. It would be _awesome_ if we could have a HashSet<ReadOnlySpan<char>> but that's not going to happen as those would be structs not objects.